mantidproject/mantid
# Extend SetUncertainties with custom mode

Fixes #15326.

Extended SetUncertainties algorithm with "custom" mode, allowing users to replace an arbitrary error value with another arbitrary value. Users can also set the precision of the comparison used for finding values to replace.

**To test:**
1. Create a new sample workspace with random values. 
1. Run SetUncertainties with "custom" mode. Set IfEqualTo to some value between 0 - 1. Pick any arbitrary value for SetErrorTo. 
1. Open the output work space and confirm (only) errors within tolerance to the chosen value (as defined by precision) have been replaced.
1.  Try different values for precision to verify it works as described in the ticket (the precision value is the amount of decimal places to check).
1. All other modes should be unchanged.

[Release notes here](https://github.com/mantidproject/mantid/commit/d75edf29496a8661732fb9b89c43d555e09831fb)


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
